Deprecate jaeger-thrift-splunk (#191)
signalfx/gdi-specification: specification/configuration.md

  - Distribution MUST NOT list Zipkin exporter as supported (not supported by Smart Agent)
- `OTEL_TRACES_EXPORTER`
  - Non-RUM distribution MUST default to `otlp` over gRPC with an endpoint of `localhost:4317`
  - Non-RUM distribution MAY offer `jaeger-thrift-splunk` that defaults to `http://127.0.0.1:9080/v1/trace`.
    **`jaeger-thrift-splunk` is deprecated.**
    If `jaeger-thrift-splunk` is selected by the user, distributions MUST log a deprecation warning proposing an alternate method, e.g.:

    ```
    jaeger-thrift-splunk trace exporter is deprecated and may be removed in a future major release.

>     We suggest that you use the default otlp exporter instead. You can set the SPLUNK_REALM and SPLUNK_ACCESS_TOKEN environment variables to send spans directly to Splunk Observabilty Cloud.

Reviewer: split the line, typo and wording proposal 😉 

```suggestion
    We suggest that you use the default otlp exporter instead.
    You can set the SPLUNK_REALM and SPLUNK_ACCESS_TOKEN environment variables to send telemetry directly to Splunk Observability Cloud.
```

  - Distribution MUST NOT list Zipkin exporter as supported (not supported by Smart Agent)
- `OTEL_TRACES_EXPORTER`
  - Non-RUM distribution MUST default to `otlp` over gRPC with an endpoint of `localhost:4317`
  - Non-RUM distribution MAY offer `jaeger-thrift-splunk` that defaults to `http://127.0.0.1:9080/v1/trace`.

>     **`jaeger-thrift-splunk` is deprecated.**
>     If `jaeger-thrift-splunk` is selected by the user, distributions MUST log a deprecation warning proposing an alternate method, e.g.:
> 
>     ```
>     jaeger-thrift-splunk trace exporter is deprecated and may be removed in a future major release.
>     We suggest that you use the default otlp exporter instead.
>     You can set the SPLUNK_REALM and SPLUNK_ACCESS_TOKEN environment variables to send telemetry directly to Splunk Observability Cloud.
>     ```

Reviewer 2: ```suggestion
    **NOTE: `jaeger-thrift-splunk` is deprecated.**
    If the user selects `jaeger-thrift-splunk`, distributions MUST log a deprecation warning and suggest an alternate method. For example:

    ```
    jaeger-thrift-splunk trace exporter is deprecated and may be removed in a future major release. Use the default 
    OTLP exporter instead, or set the SPLUNK_REALM and SPLUNK_ACCESS_TOKEN environment variables to send 
    telemetry directly to Splunk Observability Cloud.
    ```
```

  - Distribution MUST NOT list Zipkin exporter as supported (not supported by Smart Agent)
- `OTEL_TRACES_EXPORTER`
  - Non-RUM distribution MUST default to `otlp` over gRPC with an endpoint of `localhost:4317`

>   - Non-RUM distribution MAY offer `jaeger-thrift-splunk` that defaults to `http://127.0.0.1:9080/v1/trace`. **`jaeger-thrift-splunk` is deprecated. Distributions SHOULD provide a deprecation warning.**

Reviewer: Is there any reason why not using "MUST" instead of "SHOULD" regarding the deprecation warning?

Also I think we could try to reuse the wording from the scenario when `OTEL_SERVICE_NAME` is empry. Proposal:

```md
  - Non-RUM distribution MAY offer `jaeger-thrift-splunk` that defaults to `http://127.0.0.1:9080/v1/trace`.
     **`jaeger-thrift-splunk` is deprecated.
     If the user uses `jaeger-thrift-splunk` the distribution MUST log a
     warning. The warning message MUST clearly describe it and propose an alternative. E.g.

     ` ` ` (remove spaces)
     jaeger-thrift-splunk trace exporter is deprecated and may be removed in a future release.
     We suggest that you use the default otlp exporter instead. You can set the SPLUNK_REALM
     environmental variable to send spans directly to Splunk Observabilty Cloud. 
     ` ` ` (remove spaces)
```

Author: Changed it to MUST, wasn't completely sure here. Check the wording now
